virtwho/virt/virt.py

```diff
@@ -492,6 +497,34 @@ def _get_data(self):
             reports[source_key] = report
         return reports
 
+    def _get_data_initial(self):
```

Comment:
I generally like the idea although I would say I think the log messages should be a little different for the initial checks vs the regular checks. I would also say that I don't want to be checking for duplicates etc in the initial run. I could remove that logic from this loop, but then that would mean going through all the items in the list returned from the common method in the non-initial checks to remove any that a duplicate.